transport/client.go

```diff
@@ -311,7 +315,7 @@ func Dial(address string, tr *Transport) (*tls.Conn, error) {
 
 	for _, chain := range state.VerifiedChains {
 		for _, cert := range chain {
-			revoked, ok := revoke.VerifyCertificate(cert)
```

Comment:
How can they share checker if there could be different TLS confs? I.e. 444 uses /path/crl1.crl and hardfail=true (for exmaple with self signed certs ), 443 uses InCert URLs (ocsp, crl) and hardfail=false with a regular CA.